lib/Doctrine/ORM/Internal/Hydration/DqlAliasTreeWalker.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Internal\Hydration;
+
+/**
+ * @internal
+ */
+class DqlAliasTreeWalker {
```

Comment:
This class is not really a tree walker. Let's merge those two methods into `ArrayHydrator`, shall we?

Reply:
Thank you. you're right. I changed that :)